lib/src/components/NxList/__tests__/NxListButtonItem.test.tsx

```diff
@@ -5,76 +5,119 @@
-import 'jest-enzyme';
+import { NxListButtonItemProps } from '../types';
 
-describe('NxListButtonItem', function() {
+describe('NxList.ButtonItem', function() {
+  const quickRender = rtlRender(NxList.ButtonItem, {}),
+      renderEl = rtlRenderElement(NxList.ButtonItem, {});
 
-  const minimalProps: NxListButtonItemProps = {};
-  const getShallow = getShallowComponent(NxList.ButtonItem, minimalProps);
-  const getMounted = getMountedComponent(NxList.ButtonItem, minimalProps);
+  it('renders a list containing a button', function() {
+    const view = quickRender(),
+        listItem = view.getByRole('listitem'),
+        btn = view.getByRole('button');
+
+    expect(listItem).toBeInTheDocument();
+    expect(btn).toBeInTheDocument();
+    expect(listItem).toContainElement(btn);
+  });
 
-  it('renders a correctly structured element - a button within an li', function() {
-    const contentEl = getShallow();
+  it('sets a ref to the element', function() {
+    const ref = React.createRef<HTMLLIElement>(),
+        el = renderEl({ ref });
 
-    expect(contentEl).toExist();
-    expect(contentEl).toMatchSelector('.nx-list__item.nx-list__item--clickable');
-    expect(contentEl.find('button')).toMatchSelector('.nx-list__btn');
-    expect(contentEl.find('button').parent().is('li'));
+    expect(ref.current).toBe(el);
   });
 
-  it('renders the classNames given to both the li and the button', function() {
-    const contentEl = getShallow({className: 'customClassLi', buttonClassName: 'customClassBtn'});
+  it('sets the specified classnames to the list in addition to the defaults', function() {
+    const el = renderEl({ className: 'foo' }),
+        defaultEl = renderEl()!;
 
-    expect(contentEl).toExist();
-    expect(contentEl).toMatchSelector('.nx-list__item.nx-list__item--clickable.customClassLi');
-    expect(contentEl.find('button')).toMatchSelector('.nx-list__btn.customClassBtn');
+    expect(el).toHaveClass('foo');
+
+    for (const cls of Array.from(defaultEl.classList)) {
+      expect(el).toHaveClass(cls);
+    }
   });
 
-  it('renders a clickable list whose list item element has prop disabled and is disabled', function() {
-    const contentEl = getMounted({disabled: true});
+  it('sets the specified attrs to the list', function() {
+    const el = renderEl({ id: 'foo', lang: 'en' });
 
-    expect(contentEl).toExist();
-    expect(contentEl).toHaveProp('disabled');
-    expect(contentEl).toBeDisabled();
+    expect(el).toHaveAttribute('id', 'foo');
+    expect(el).toHaveAttribute('lang', 'en');
   });
 
-  it('renders a clickable list whose list item element has prop selected and is selected', function() {
-    const contentEl = getMounted({selected: true});
+  it('sets the specified classnames to the button in addition to the defaults ' +
+    'when buttonClassName prop is provided', function() {
+    const btn = quickRender({ buttonClassName: 'boo' }).getByRole('button'),
```

Comment:
Since we have two classname related props we should test that each of them affects only the element that it's supposed to and not the other.  That is, in this test also set a `className` and assert that it doesn't end up on the button, and in 'sets the specified classnames to the list in addition to the defaults', add a `buttonClassName` and assert that it doesn't end up on the listitem.

Reply:
Updated here 22441387eb16d63f3de01de20428262afc12a5ed